// frontend/src/pages/ApercuHistorique.jsx
import React, { useEffect } from "react";
import { useTranslation } from "react-i18next";
import PageSidebar from "../components/PageSidebar";
import ap1 from "../assets/ap1.png";
import ap2 from "../assets/ap2.png";
import ap3 from "../assets/ap3.png";

const SLUG = "apercu-historique";

export default function ApercuHistorique() {
  const { i18n } = useTranslation();
  const lang = i18n.language;
  const isAr = lang === 'ar';
  const dir = isAr ? 'rtl' : 'ltr';

  useEffect(() => {
    const hEl = document.querySelector('.site-header, header, .main-nav, #mainNav');
    if (hEl) {
      document.documentElement.style.setProperty('--header-height', hEl.getBoundingClientRect().height + 'px');
    }
  }, [lang]);

  const fr = {
    title: 'Aperçu Historique',
    intro: 'EL HAJEB aurait tiré son nom actuel de deux légendes :',
    b1: 'Historiquement, cette Province constituait une ceinture protectrice de la capitale Ismaïlite.',
    b2: 'Géographiquement, le relief montagneux d\'EL HAJEB, sculpté par les effets de l\'érosion, a toujours abrité des sources naturelles ("Hajbes") dont les deux principales : "Aïn Madani" et "Aïn Khadem".',
    c1:'Source naturelle',c2:'Source Aïn El Madani',c3:'Source Aïn Khadem'
  };
  const ar = {
    title: 'لمحة تاريخية',
    intro: 'يقال إن مدينة الحاجب استمدت اسمها الحالي من أسطورتين:',
    b1: 'تاريخياً شكّلت هذه المنطقة حزاماً واقياً للعاصمة الإسماعيلية.',
    b2: 'جغرافياً، ظلّ التضاريس الجبلية للحاجب، التي شكّلتها عوامل التعرية، تحتضن ينابيع طبيعية (الحواجب) من أهمها: عين المدني و عين خادم.',
    c1:'منبع طبيعي',c2:'عين المدني',c3:'عين خادم'
  };
  const t = isAr ? ar : fr;

  return (
    <div className="pg-shell geo-static-shell" dir={dir}>
      <ApercuStyle />
      <div className="geo-layout">
        <main className="geo-main">
          <header className="geo-head">
            <h1 className="geo-title">{t.title}</h1>
            <div className="geo-sep" />
          </header>

          <article className="geo-article" dir={dir} id="apercu-historique">
            <p className="geo-p geo-intro">{t.intro}</p>
            <ul className="geo-list">
              <li>{t.b1}</li>
              <li>{t.b2}</li>
            </ul>
            <div className="ap-gallery">
              {[{src:ap1,cap:t.c1},{src:ap2,cap:t.c2},{src:ap3,cap:t.c3}].map((im,i)=>(
                <figure className="ap-fig" key={i}>
                  <div className="ap-media"><img src={im.src} alt={im.cap} loading="lazy" /></div>
                  <figcaption>{im.cap}</figcaption>
                </figure>
              ))}
            </div>
          </article>
        </main>
        <aside className="geo-side"><PageSidebar lang={lang} slug={SLUG} theme="light" /></aside>
      </div>
    </div>
  );
}

function ApercuStyle() {
  return (
    <style>{`
  .pg-shell.geo-static-shell {
    min-height: 100vh;
    padding: 4.5rem clamp(1rem,3vw,3rem) 6rem;
    background:
      radial-gradient(circle at 12% 18%,#ecfdf5,transparent 65%),
      radial-gradient(circle at 85% 82%,#d1fae5,transparent 70%),
      linear-gradient(125deg,#ffffff,#f0fdf4 55%,#e6fcf3);
  }
  .geo-layout{display:grid;grid-template-columns:1fr;gap:3.2rem;max-width:1350px;margin:0 auto;}
  @media (min-width:1280px){.geo-layout{grid-template-columns:minmax(0,1fr) 330px;}}
  .geo-head{margin:0 0 1.6rem;}
  .geo-title{margin:0;font-size:clamp(2.2rem,4.5vw,3.25rem);font-weight:800;line-height:1.05;background:linear-gradient(90deg,#047857,#059669 35%,#10b981 70%);-webkit-background-clip:text;color:transparent;}
  .geo-sep{height:4px;width:100%;margin-top:.9rem;background:linear-gradient(90deg,transparent,#10b981,#34d399 70%,transparent);border-radius:2rem;}
  
  .geo-article{position:relative;background:#fff;border:1px solid #e5e7eb;border-radius:2rem;padding:clamp(1.5rem,2.4vw,2.7rem);box-shadow:0 6px 18px rgba(0,0,0,0.08);}
  .geo-p{font-size:1rem;line-height:1.65;color:#1f2937;font-weight:500;margin:0 0 1rem;}
  .geo-intro{font-weight:700;color:#065f46;}
  .geo-list{margin:0 0 1.6rem;padding-inline-start:1.3rem;display:grid;gap:.55rem;color:#065f46;font-weight:500;}
  [dir='rtl'] .geo-list{padding-inline-start:0;padding-inline-end:1.3rem;}
  .geo-list li::marker{color:#10b981;}
  
  .ap-gallery{display:grid;grid-template-columns:repeat(auto-fit,minmax(280px,1fr));gap:1.5rem;margin-top:2.6rem;}
  .ap-fig{margin:0;display:flex;flex-direction:column;gap:.5rem;}
  .ap-media{position:relative;aspect-ratio:16/9;overflow:hidden;border-radius:1.1rem;box-shadow:0 6px 20px -8px rgba(0,0,0,.18);background:#f1f5f9;}
  .ap-media img{width:100%;height:100%;object-fit:cover;display:block;}
  .ap-fig figcaption{font-size:.65rem;letter-spacing:.14em;text-transform:uppercase;font-weight:600;color:#065f46;text-align:center;}
  
  @media (max-width:900px){
    .pg-shell.geo-static-shell{padding:4.5rem 1rem 5rem;}
    .geo-article{padding:1.4rem 1.2rem 1.9rem;}
  .ap-gallery{gap:1rem;margin-top:2.2rem;}
  .ap-fig{gap:.45rem;}
  .ap-media{border-radius:1rem;}
  }
  `}</style>
  );
}
