// frontend/src/pages/SituationGeographique.jsx
import React, { useEffect } from "react";
import { useTranslation } from "react-i18next";
import PageSidebar from "../components/PageSidebar";
import carteImg from "../assets/carte.png";

const MAP_SRC =
  "https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d33511.56630540291!2d-5.3717392!3d33.682252749999996!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0xda0353572c48e53%3A0x421b89bdd8f41877!2sEl%20Hajeb!5e1!3m2!1sen!2sma!4v1755372935139!5m2!1sen!2sma";

const SLUG = "situation-geographique";

export default function SituationGeographique() {
  const { i18n } = useTranslation();
  const lang = i18n.language;
  const isAr = lang === "ar";
  const dir = isAr ? "rtl" : "ltr";

  useEffect(() => {
    const hEl = document.querySelector(
      ".site-header, header, .main-nav, #mainNav"
    );
    if (hEl) {
      document.documentElement.style.setProperty(
        "--header-height",
        hEl.getBoundingClientRect().height + "px"
      );
    }
  }, [lang]);

  const title = isAr ? "الموقع الجغرافي" : "Situation Géographique";

  const fr = {
    title: "Situation Géographique",
    intro1:
      "La Province d'El Hajeb a été créée par le Décret Royal n° 2.91.90 du 1er janvier 1991. Elle est l'une des neuf provinces et préfectures formant la région Fès-Meknès.",
    delim: "La province est délimitée par :",
    borders: [
      "La préfecture de Meknès au nord",
      "La province d'Ifrane et la province de Khénifra au sud",
      "La préfecture de Fès et la province de Sefrou à l'est",
      "La province de Khémisset à l'ouest",
    ],
    roleTitle: "Rôle et Importance",
    role1:
      "Compte tenu de son histoire et de sa géographie, la province d'El Hajeb a joué et continue de jouer un rôle important dans le processus de développement du Maroc moderne et dans sa marche vers le progrès.",
    role2:
      "Sa position stratégique au centre de la région Fès-Meknès place ce territoire au cœur des dynamiques de développement régional et national.",
    mapAlt: "Carte de localisation El Hajeb",
  };

  const ar = {
    title: "الموقع الجغرافي",
    intro1:
      "أُحدِث إقليم الحاجب بمرسوم ملكي رقم 2.91.90 بتاريخ 1 يناير 1991. وهو واحد من تسعة أقاليم وعمالات التي تُكوّن جهة فاس - مكناس.",
    delim: "يحد الإقليم:",
    borders: [
      "عمالة مكناس شمالاً",
      "إقليم إفران وإقليم خنيفرة جنوباً",
      "عمالة فاس وإقليم صفرو شرقاً",
      "إقليم الخميسات غرباً",
    ],
    roleTitle: "الدور والأهمية",
    role1:
      "نظراً لتاريخه وموقعه الجغرافي لعب إقليم الحاجب وما يزال دوراً مهماً في مسار التنمية بالمغرب الحديث وفي مسيرته نحو التقدم.",
    role2:
      "إن تموقعه الإستراتيجي بوسط جهة فاس - مكناس يجعل منه مجالاً محورياً في ديناميات التنمية الجهوية والوطنية.",
    mapAlt: "خريطة موقع الحاجب",
  };

  const t = isAr ? ar : fr;

  return (
    <div className="pg-shell geo-static-shell" dir={dir}>
      <GeoStyle />
      <div className="geo-layout">
        <main className="geo-main">
          <header className="geo-head">
            <h1 className="geo-title">{title}</h1>
            <div className="geo-sep" />
          </header>

          <article className="geo-article" dir={dir}>
            {/* Map FIRST */}
            <div className="geo-map-wrap">
              <div className="ratio">
                <iframe
                  title={t.mapAlt}
                  src={MAP_SRC}
                  loading="lazy"
                  allowFullScreen
                  referrerPolicy="no-referrer-when-downgrade"
                />
              </div>
            </div>

            {/* Then Text */}
            <p className="geo-p">{t.intro1}</p>
            <p className="geo-p geo-delim">{t.delim}</p>
            <ul className="geo-list">
              {t.borders.map((b, i) => (
                <li key={i}>{b}</li>
              ))}
            </ul>
            <h2 className="geo-sub">{t.roleTitle}</h2>
            <p className="geo-p">{t.role1}</p>
            <p className="geo-p">{t.role2}</p>

            {/* Image LAST */}
            <figure className="geo-fig">
              <img src={carteImg} alt={t.mapAlt} className="geo-img" />
              <figcaption className="geo-cap">
                {isAr
                  ? "خريطة تقريبية لموقع الإقليم"
                  : "Carte illustrative de la province"}
              </figcaption>
            </figure>
          </article>
        </main>
        <aside className="geo-side">
          <PageSidebar lang={lang} slug={SLUG} theme="light" />
        </aside>
      </div>
    </div>
  );
}

function GeoStyle() {
  return (
    <style>{`
  .pg-shell.geo-static-shell {
    min-height: 100vh;
    padding: 4.5rem clamp(1rem,3vw,3rem) 6rem;
    background:
      radial-gradient(circle at 12% 18%,#ecfdf5,transparent 65%),
      radial-gradient(circle at 85% 82%,#d1fae5,transparent 70%),
      linear-gradient(125deg,#ffffff,#f0fdf4 55%,#e6fcf3);
  }
  .geo-layout{display:grid;grid-template-columns:1fr;gap:3.2rem;max-width:1350px;margin:0 auto;}
  @media (min-width:1280px){.geo-layout{grid-template-columns:minmax(0,1fr) 330px;}}
  .geo-head{margin:0 0 1.6rem;}
  .geo-title{margin:0;font-size:clamp(2.2rem,4.5vw,3.25rem);font-weight:800;line-height:1.05;background:linear-gradient(90deg,#047857,#059669 35%,#10b981 70%);-webkit-background-clip:text;color:transparent;}
  .geo-sep{height:4px;width:100%;margin-top:.9rem;background:linear-gradient(90deg,transparent,#10b981,#34d399 70%,transparent);border-radius:2rem;}
  
  .geo-article{position:relative;background:#fff;border:1px solid #e5e7eb;border-radius:2rem;padding:clamp(1.5rem,2.4vw,2.7rem);box-shadow:0 6px 18px rgba(0,0,0,0.08);}
  .geo-p{font-size:1rem;line-height:1.65;color:#1f2937;font-weight:500;margin:0 0 1rem;}
  .geo-delim{font-weight:700;color:#065f46;}
  .geo-list{margin:0 0 1.6rem;padding-inline-start:1.3rem;display:grid;gap:.55rem;color:#065f46;font-weight:500;}
  [dir='rtl'] .geo-list{padding-inline-start:0;padding-inline-end:1.3rem;}
  .geo-list li::marker{color:#10b981;}
  
  .geo-sub{font-size:clamp(1.45rem,2.2vw,2.05rem);font-weight:800;margin:2.2rem 0 1rem;background:linear-gradient(90deg,#065f46,#10b981);-webkit-background-clip:text;color:transparent;}
  
  .geo-map-wrap{margin:2.4rem 0 2.6rem;}
  .ratio{position:relative;width:100%;aspect-ratio:16/9;border-radius:1.4rem;overflow:hidden;background:#f9fafb;box-shadow:0 6px 20px rgba(0,0,0,.12);}
  .ratio iframe{position:absolute;inset:0;width:100%;height:100%;border:0;}
  
  .geo-fig{margin:2.8rem auto 0;max-width:840px;text-align:center;}
  .geo-img{max-width:100%;width:100%;border-radius:1.2rem;box-shadow:0 8px 22px rgba(0,0,0,.15);display:block;margin:0 auto;}
  .geo-cap{font-size:.7rem;letter-spacing:.12em;text-transform:uppercase;font-weight:600;color:#065f46;margin-top:.9rem;}
  
  @media (max-width:900px){.pg-shell.geo-static-shell{padding:4.5rem 1rem 5rem;} .geo-article{padding:1.4rem 1.2rem 1.9rem;} .ratio{border-radius:1rem;}}
  `}</style>
  );
}
